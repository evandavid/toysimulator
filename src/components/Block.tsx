import React from 'react';
import {View} from 'react-native';
import styled from 'styled-components';

const BlockWrapper = styled(View)<{color?: string; height: number}>`
  width: 100%;
  height: ${props => props.height}px;
  background-color: ${props => props.color};
`;

type BlockProps = {
  index: number;
  height: number;
};

const Block = ({index, height}: BlockProps) => {
  const colors = [
    '#FFB6E8',
    '#FF9CEE',
    '#FFCCF9',
    '#FCC2FF',
    '#F6A6FF',
    '#B28DFF',
    '#C5A3FF',
    '#D5AAFF',
    '#ECD4FF',
    '#FBE4FF',
    '#DCD3FF',
    '#A79AFF',
    '#B5B9FF',
    '#97A2FF',
    '#AFCBFF',
    '#AFF8D8',
    '#C4FAF8',
    '#85E3FF',
    '#ACE7FF',
    '#6EB5FF',
    '#FFC9DE',
    '#DBFFD6',
    '#F3FFE3',
    '#E7FFAC',
    '#FFFFD1',
  ];

  return <BlockWrapper height={height} color={colors[index]} />;
};

export default Block;
