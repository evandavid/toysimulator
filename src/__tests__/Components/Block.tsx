import React from 'react';
import renderer from 'react-test-renderer';
import Block from '../../components/Block';

it('renders correctly', () => {
  const tree = renderer.create(<Block index={1} height={60} />).toJSON();
  expect(tree).toMatchSnapshot();
});
