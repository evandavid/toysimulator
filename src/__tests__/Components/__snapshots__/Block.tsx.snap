// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="sc-dlfnuX kGQTrV"
  color="#FF9CEE"
  height={60}
/>
`;
